views/messages.js

```diff
@@ -4,13 +4,14 @@ var readonly = require('read-only-stream')
 var charwise = require('charwise')
 var xtend = require('xtend')
 var EventEmitter = require('events').EventEmitter
+var isValidMessage = require("../util").isValidMessage
 
 module.exports = function (lvl) {
   var events = new EventEmitter()
 
   return View(lvl, {
     map: function (msg) {
-      if (!msg.value.timestamp) return []
+      if (!isValidMessage || !msg.value.timestamp) return []
```

Comment:
`s/isValidMessage/isValidMessage(msg)`

strongly recommend writing tests for this new logic!